core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -0,0 +1,131 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.Nullable;
+
+import com.google.android.exoplayer2.Player;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.ads.AdPlaybackState;
+import com.google.android.exoplayer2.source.ads.AdsLoader;
+import com.novoda.noplayer.AdvertBreak;
+import com.novoda.noplayer.AdvertsLoader;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class NoPlayerAdsLoader implements AdsLoader, Player.EventListener {
```

Comment:
The purpose of this class is to load the Adverts and then maintain the state of `AdPlaybackState` in order to ensure the ads are playing properly.